src/common/pf_snmp.c

```diff
@@ -0,0 +1,440 @@
+   }
+   else
+   {
+      LOG_INFO (
+         PF_SNMP_LOG,
+         "SNMP(%d): Did read sysContact from nvm.\n",
+         __LINE__);
+   }
+   contact->string[sizeof (contact->string) - 1] = '\0';
+}
+
+int pf_snmp_set_system_contact (
+   pnet_t * net,
+   const pf_snmp_system_contact_t * contact)
+{
+   const char * directory = net->fspm_cfg.file_directory;
+   pf_snmp_system_contact_t temporary_buffer;
+   int res;
+
+   res = pf_file_save_if_modified (
+      directory,
+      PNET_FILENAME_SYSCONTACT,
+      (void *)contact, /* TODO: Remove cast after adding 'const' to parameter
+                        */
+      &temporary_buffer,
+      sizeof (*contact));
+   switch (res)
+   {
+   case 2:
+      LOG_INFO (
+         PF_SNMP_LOG,
+         "SNMP(%d): First nvm saving of sysContact. "
+         "sysContact: %s\n",
+         __LINE__,
+         contact->string);
+      break;
+   case 1:
+      LOG_INFO (
+         PF_SNMP_LOG,
+         "SNMP(%d): Updating nvm stored sysContact. "
+         "sysContact: %s\n",
+         __LINE__,
+         contact->string);
+      break;
+   case 0:
+      LOG_INFO (
+         PF_SNMP_LOG,
+         "SNMP(%d): No storing of nvm sysContact (no changes). "
+         "sysContact: %s\n",
+         __LINE__,
+         contact->string);
+      break;
+   default:
+      /* Fall-through */
+   case -1:
+      LOG_ERROR (
+         PF_SNMP_LOG,
+         "SNMP(%d): Failed to store nvm sysContact.\n",
+         __LINE__);
+      break;
+   }
+
+   return res == -1 ? -1 : 0;
+}
+
+void pf_snmp_get_system_location (
+   pnet_t * net,
+   pf_snmp_system_location_t * location)
+{
+   snprintf (
+      location->string,
+      sizeof (location->string),
+      net->fspm_cfg.im_1_data.im_tag_location);
+}
+
+int pf_snmp_set_system_location (
+   pnet_t * net,
+   const pf_snmp_system_location_t * location)
+{
+   /* TODO: Write to I&M1 */
+   return -1;
+}
+
+void pf_snmp_get_system_description (
+   pnet_t * net,
+   pf_snmp_system_description_t * description)
+{
+   /* Encode as per Profinet specification:
+    * DeviceType, Blank, OrderID, Blank, IM_Serial_Number, Blank, HWRevision,
+    * Blank, SWRevisionPrefix, SWRevision.
+    *
+    * See IEC CDV 61158-6-10 (PN-AL-Protocol) ch. 5.1.2 "APDU abstract syntax",
+    * field "SystemIdentification".
+    */
+   snprintf (
+      description->string,
+      sizeof (description->string),
+      "%-" STRINGIFY (
+         PNET_PRODUCT_NAME_MAX_LEN) "s "
+                                    "%-" STRINGIFY (
+                                       PNET_ORDER_ID_MAX_LEN) "s "
+                                                              "%-" STRINGIFY (
+                                                                 PNET_SERIAL_NUMBER_MAX_LEN) "s "
+                                                                                             "%5u "
+                                                                                             "%c%3u%3u%3u",
```

Comment:
clang-format does something weird here. Maybe adjust a .clang-format setting?

Reply:
That might be difficult to fix. Wrap the code in /* clang-format-off */ ... /* clang-format-on '/ to disable formatting for the region.

Reply:
Done.